hanabi_lib/canonical_encoders.cc

```diff
@@ -421,8 +437,16 @@ int EncodeLastAction_(const HanabiGame& game,
   return offset - start_offset;
 }
 
-int CardKnowledgeSectionLength(const HanabiGame& game) {
-  return game.NumPlayers() * game.HandSize() *
+int CardKnowledgeSectionLength(const HanabiGame& game, bool using_joint_obs) {
```

Comment:
card_knowledge_length can v0_belief_length should be exactly the same. Why do we need 2 functions here?

Reply:
This way we can keep intact the logic of `EncodeV0Belief`, which relies on the length of `EncodeCardKnowledge`, and then just pad zeros at the end of `EncodeV0Belief` for all the additional players. Seems like the safest way to encode the joint belief.